src/docfx/build/dependency/DependencyResolver.cs

```diff
@@ -36,7 +36,7 @@ public DependencyResolver(GitCommitProvider gitCommitProvider, BookmarkValidator
 
         // forLandingPage should not be used, it is a hack to handle some specific logic for landing page based on the user input for now
         // which needs to be removed once the user input is correct
-        public (Error error, string link, Document file) ResolveLink(string path, Document relativeTo, Document resultRelativeTo, Action<Document> buildChild, bool forLandingPage = false)
+        public (Error error, string link, Document file) ResolveLink(string path, Document relativeTo, Document resultRelativeTo, Action<Document> buildChild, Range range, bool forLandingPage = false)
```

Comment:
```suggestion
        public (Error error, string link, Document file) ResolveLink(string path, Document relativeTo, Document resultRelativeTo, Action<Document> buildChild, in Range range, bool forLandingPage = false)
```

Reply:
why do we need reference input here? `range` is not in use for the caller

Reply:
`Range` is currently a `struct` and `in` makes it pass by ref instead of copying the whole value.

Reply:
what's the benefit to make it as a struct? it's used a lot in Errors.cs as parameters

Reply:
> `Range` is currently a `struct` and `in` makes it pass by ref instead of copying the whole value.

Ah, I remember now :)